### PR TITLE
tools: remove the git status check in 'make check'

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve? 
Before: 
```shell
gofmt (simplify)
errcheck
linting
go mod tidy
./tools/check/check-tidy.sh
make: *** [Makefile:118: tidy] Error 1
```
Seeing such output, developers don't know what went wrong without looking into the shell script.

After this PR:
```bash
gofmt (simplify)
errcheck
linting
go mod tidy
./tools/check/check-tidy.sh
The git working tree is not clean.
make: *** [Makefile:118: tidy] Error 1
```
__The git working tree is not clean.__
This is much better.

### What is changed and how it works?
- Modify tools/check/check-tidy.sh.
- Add some comments for the script.

### Check List 

Tests 

 - Manual test

Code changes

N/A

Side effects

N/A

Related changes

N/A

Release note

N/A

### CODE REVIEW

File: tools/check/check-tidy.sh [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
+#       or to zero if all commands of the pipeline exit successfully.
 set -euo pipefail
 
 # go mod tidy do not support symlink
 cd -P .
 
 GO111MODULE=on go mod tidy
-git diff --quiet
+
+git diff --quiet || { echo "The git working tree is not clean." && exit 1; }
```
Comment (jackysp) [inline tools/check/check-tidy.sh]: I don't know why we need `git diff --quiet`. Should we remove it?

Comment (tangenta) [reply tools/check/check-tidy.sh]: @jackysp It ensures that all the changes we have made are committed.

Comment (jackysp) [reply tools/check/check-tidy.sh]: I think it is not necessary. It forces us to commit all changes before `make check`. But `make check` may encounter a format issue, which we have to commit the changes again.

**Comment by Deardrops** [comment]
Suggestion:
```
The git working tree is not clean. please make sure you have commited all local changes.
```

File: tools/check/check-tidy.sh (hunk 1 above)
Comment (tangenta) [reply tools/check/check-tidy.sh]: @jackysp Updated. PTAL

**Comment by jackysp** [comment]
PTAL @tiancaiamao

**Comment by jackysp** [comment]
PTAL @tiancaiamao

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**